### PR TITLE
Update intro and cbor quick start in README

### PR DESCRIPTION
**Pull request by fxamacker**
Updates #420 

Updated README:
- Update intro and quick start
- Removed content and sections such as "Usage" because it was too redundant or outdated.

Content above the badges is for summary posted on go.dev.

### CODE REVIEW

**Comment by fxamacker** [comment]
@x448 Can you take a look when you have a moment? :pray:

File: README.md [hunk 1]
```diff
@@ -18,145 +20,214 @@ Install with `go get github.com/fxamacker/cbor/v2` and `import "github.com/fxama
...
+Features include CBOR [extension points](https://www.rfc-editor.org/rfc/rfc8949.html#section-7.1) (e.g. CBOR tags) and extensive settings.  API has interfaces that allow users to create custom encoding and decoding without modifying this library.
+
+</details>
+
+<details><summary>Efficient Rejection of Malicious CBOR Data</summary><p/>
+
+Decoding 10 bytes of malicious data into `[]byte` doesn't exhaust memory. E.g.  
+`[]byte{0x9B, 0x00, 0x00, 0x42, 0xFA, 0x42, 0xFA, 0x42, 0xFA, 0x42}`.
+
+| Codec | Speed (ns/op) | Memory | Allocs |
```
Comment (x448) [inline README.md:68]: This table should be updated.  I can try on my laptop.

File: README.md [hunk 2]
```diff
@@ -1097,8 +401,6 @@ To prevent delays to release schedules, fuzzing is not restarted for a release i
 
 <hr>
 
-⚓  [Quick Start](#quick-start) • [Features](#features) • [Standards](#standards) • [API](#api) • [Options](#options) • [Usage](#usage) • [Fuzzing](#fuzzing-and-code-coverage) • [License](#license)
-
 ## Versions and API Changes
```
Comment (x448) [inline README.md:404]: Remove ", etc." in this sentence to make it less broad.  Maybe sentence isn't necessary?

> Bug fixes like detecting an error that was missed in prior version are excluded from SemVer as long as function parameters, etc. are unchanged.

**Review by x448 — APPROVED** [review]
Amazing work!  Quick start looks great!

Design and Feature Highlights looks a bit weird.  Maybe too long or something else?

**Review by x448 — APPROVED** [review]
Fantastic improvement!

**Pull request merged**